### PR TITLE
Language and description of Assignment 4a

### PR DESCRIPTION
**Pull request by shiffman**
Here is a write-up based on class discussion, feedback appreciated? Is this too many questions to address? And I missing any important ones?

### CODE REVIEW

**Comment by shiffman** [comment]
Also @lydiajessup to go along with this I quickly created this Dataset wiki:

https://github.com/ml5js/Intro-ML-Arts-IMA/wiki/Datasets

I copy pasted the google doc, but didn't have time to go through and format / add all the links yet. I wonder if there is a nice way to export from google doc to markdown? I

File: 04_diy_neural/README.md [hunk 1]
```diff
@@ -34,7 +34,19 @@
...
 
 ### Assignment 4A due Wednesday, Sept 24, 9am
-* TBA (find and link to a dataset)
+1. Find a dataset that interests you. Some ideas:
+    * Something you find online. For example, take a look at [Kaggle](https://www.kaggle.com/) and you can find a list of datasets we've compiled on the [dataset wiki](https://github.com/ml5js/Intro-ML-Arts-IMA/wiki/Datasets).
+    * Find aa dataset that you collect yourself or is already being collected about you. For example, personal data like steps taken per day, browser history, minutes spent on your mobile device, sensor readings, and more.
+    * A dataset that you collect by crowdsourcing data. One way to approach this is to create a google form with questions and ask friends and fellow students to complete the form.
+2. After you have found or collected your dataset, write a blog post that addresses these questions:
+    * What is (are!) the data?
+    * What format is the data? (CSV, JSON, PDF, or . . )
```
Comment (lydiajessup) [inline 04_diy_neural/README.md]: just a typo here I think - should be: What format is the data in?

File: 04_diy_neural/README.md [hunk 2]
```diff
@@ -34,7 +34,19 @@
...
+    * What are the dimensions of the data (rows and columns) and what are the data types (numbers, strings, images, etc.)?
+    * Is there missing, incorrect, or otherwise problematic data?
+    * How and why was this data collected?
+    * For whom is this data accurate or useful? What is this data *unrepresentative* of? (Who is missing and left out of the data?)
+    * Is this data appropriate for Machine Learning
```
Comment (lydiajessup) [inline 04_diy_neural/README.md]: missing ? at the end. Perhaps also clarify what appropriate means here - something like, are there other (simpler) methods that could be used to answer the question you want to ask or achieve the same result? Are there ethical concerns to using ML on the type of data you have found?

File: 04_diy_neural/README.md [hunk 3]
```diff
@@ -34,7 +34,19 @@
...
 ### Assignment 4A due Wednesday, Sept 24, 9am
-* TBA (find and link to a dataset)
+1. Find a dataset that interests you. Some ideas:
+    * Something you find online. For example, take a look at [Kaggle](https://www.kaggle.com/) and you can find a list of datasets we've compiled on the [dataset wiki](https://github.com/ml5js/Intro-ML-Arts-IMA/wiki/Datasets).
+    * Find aa dataset that you collect yourself or is already being collected about you. For example, personal data like steps taken per day, browser history, minutes spent on your mobile device, sensor readings, and more.
+    * A dataset that you collect by crowdsourcing data. One way to approach this is to create a google form with questions and ask friends and fellow students to complete the form.
+2. After you have found or collected your dataset, write a blog post that addresses these questions:
+    * What is (are!) the data?
+    * What format is the data? (CSV, JSON, PDF, or . . )
+    * What are the dimensions of the data (rows and columns) and what are the data types (numbers, strings, images, etc.)?
```
Comment (lydiajessup) [inline 04_diy_neural/README.md]: Should we explicitly ask what the variables are? It sometimes helps me better understand what I'm working with when I have to go through each one. But maybe that's too much to add to this list at the moment!

**Review by lydiajessup — COMMENTED** [review]
This looks good to me - I don't think it's too many questions. Some are quick to answer!

**Comment by lydiajessup** [comment]
@shiffman re the dataset list in the wiki - I found some tools online that can be installed or run inside of the google doc scripts but they make me nervous because they access my email or request permission to view my documents and send notifications. I'll keep looking and if I can't find it I don't mind formatting it by hand. Or perhaps linking to the google doc in the wiki for now?

File: 04_diy_neural/README.md [hunk 4]
```diff
@@ -34,7 +34,19 @@
 * Video: [Data visualization for machine learning](https://vimeo.com/304131671)
 
 ### Assignment 4A due Wednesday, Sept 24, 9am
-* TBA (find and link to a dataset)
+1. Find a dataset that interests you. Some ideas:
+    * Something you find online. For example, take a look at [Kaggle](https://www.kaggle.com/) and you can find a list of datasets we've compiled on the [dataset wiki](https://github.com/ml5js/Intro-ML-Arts-IMA/wiki/Datasets).
+    * Find aa dataset that you collect yourself or is already being collected about you. For example, personal data like steps taken per day, browser history, minutes spent on your mobile device, sensor readings, and more.
```
Comment (ellennickles) [inline 04_diy_neural/README.md]:
```suggestion
    * Find a dataset that you collect yourself or is already being collected about you. For example, personal data like steps taken per day, browser history, minutes spent on your mobile device, sensor readings, and more.
```

**Comment by ellennickles** [comment]
@lydiajessup happy to help with the formatting of the datasets wiki page if you don't find a tool! I can start from the bottom of the page and move up to meet you in the middle

File: 04_diy_neural/README.md (hunk 2 above)
Comment (ellennickles) [inline 04_diy_neural/README.md]:
```suggestion
    * Knowing what you know now about machine learning, what will a model trained on this data help you do? Are there are alternative (non-machine learning) methods you could use instead? 
```

File: 04_diy_neural/README.md (hunk 1 above)
Comment (shiffman) [reply 04_diy_neural/README.md]: thank you! fixed with https://github.com/ml5js/Intro-ML-Arts-IMA/pull/43/commits/2387e8fcaeade220faf15d3956d58fd9fbb39a4e

File: 04_diy_neural/README.md (hunk 2 above)
Comment (shiffman) [reply 04_diy_neural/README.md]: Ooops, I didn't realize I hadn't finished writing this question! Taking @ellennickles suggestion below, thank you for the excellent and thoughtful language.

File: 04_diy_neural/README.md (hunk 3 above)
Comment (shiffman) [reply 04_diy_neural/README.md]: Yes, I agree! I didn't use the term "variables" b/c I find it a little confusing in the context of programming. But it's important here. How about the following?

* What are the dimensions of the data (rows and columns)?
* What are the "variables" (also known as "data items"). In a CSV these would be the column headings. Do you recognize the data types (numbers, strings, images, etc.)?

**Comment by shiffman** [comment]
Thanks for the feedback! Merging now for the students but additional feedback, edits, clarification is welcome!

@lydiajessup I'm not sure how my brain missed the very obvious solution (even if temporary) of linking to the google doc from the wiki. I'm fine with doing it by hand and will likely find some time tomorrow if no one else has a chance to first!

**Pull request merged**